.github/workflows/build.yml

```diff
@@ -9,24 +9,46 @@ on:
 jobs:
   generate:
     runs-on: ubuntu-latest
-
     strategy:
       matrix:
         node-version: [12.x]
-
     steps:
     - uses: actions/checkout@v1
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - name: npm install, build and lint
-      run: |
-        npm ci
-        npm start
-        npm run lint
+    - name: Install dependencies
+      run: npm ci
+    - name: Generate types
+      run: npm start
+    - name: Lint types
+      run: npm run lint
       env:
-        CI: true
+        GAPI_MAX_PARALLEL: 3
+    - name: Copy discovery service response
+      if: failure()
+      run: cp .tmp/${{ env.FAILED_TYPE }}.json types/${{ env.FAILED_TYPE }}
+    - name: Upload failed type
+      uses: actions/upload-artifact@v1
+      if: failure()
+      with:
+        name: ${{ env.FAILED_TYPE }}
+        path: types/${{ env.FAILED_TYPE }}
```

Comment:
These new steps only [run on failure](https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#failure), leverage the environment variable set in `lint.ts`, and upload the relevant type directory that failed.

I [simulated this](https://github.com/namoscato/google-api-typings-generator/runs/535218548?check_suite_focus=true) on my forked repository:

<img width="418" alt="Screen Shot 2020-03-25 at 9 48 58 PM" src="https://user-images.githubusercontent.com/847532/77601905-756d8f00-6ee2-11ea-92cd-61d5fa51459b.png">